AK/Concepts.h

```diff
@@ -73,6 +76,11 @@ concept IterableContainer = requires
 // clang-format on
 }
 
+namespace AK::Detail {
+template<Concepts::AnyString T, Concepts::AnyString U>
+inline constexpr bool IsHashCompatible<T, U> = true;
```

Comment:
I'm not sure Concepts.h is the right place for this because this isn't a concept. Does it work if we define this in String.h or StringUtils.h?